### ISSUE TITLE
Can I make it such that each of the 3 buttons go to each profile?

### ISSUE THREAD

**Issue by lucaszanella** [fin-ger/logitech-m720-config#8]
Is there a way to the first side button activate profile 1, second activate profile 2 and third profile 3, instead of using profile 3 button to alternate between profiles?

**Comment by fin-ger** [comment]
I'm afraid I don't know of any way to set the mouse profile programmatically. If you are able to find a way to set the mouse profile programmatically, you might be able to do this with a Gesture detection client such as [Easystroke](https://github.com/thjaeger/easystroke/wiki) by binding a click to one of the 0x53 or 0x56 (button 8 and 9) to a shell command.

However, this still leaves you with the third button, which does not have a mapping in solaar and I don't think the mouse fires a click event for it. So you won't be able to detect whether it is clicked or not. Further, I don't think you will be able to prevent the default profile rotation behavior from happening, at least I don't know how to achieve this.

TL;DR: I think this is impossible, but I am happily proven wrong on this one!

**Comment by lucaszanella** [comment]
that's bad, I keep having to look at the mouse to see to which profile it goes. The logitech keyboard K380 has buttons for each profile and I can even deactivate them with a script.

**Comment by fin-ger** [comment]
I think the mouse is incapable of doing this, I just tested configuring this with Piper (which I would recommend, as it is far more user friendly than this script :D) and setting the profile programmatically was not possible. Do you know if this is possible in other operating systems?

**Comment by lucaszanella** [comment]
For someone like me who minimizes things running in the host machine (I only program in VMs) this script is preferable to the complexity of Piper

**Comment by fin-ger** [comment]
Hmm, I understand. But still, I don't think the mouse is capable of doing what you want. I also think I misunderstood what you meant with profiles. I thought that you meant the switching between the 2 wifi receivers and the bluetooth mode. Other logitech mice are capable of defining real profiles with buttons mappings and such. The M720 is not capable of doing this. I think you have to get a higher budget mouse for doing such things :disappointed:

**Comment by lucaszanella** [comment]
No problem, I think there's no way. Also I think you meant 2 bluetooth receivers + usb dongle. But it looks like I was able to use the 3 profiles for bluetooth, I didn't even use the dongle

**Comment by fin-ger** [comment]
The M720 has 2 unifying receiver channels and 1 bluetooth channel. I don't know how many USB dongles were shipped with the mouse, but I have several of them here and am actually using both unifying receivers right now :smile: 

> But it looks like I was able to use the 3 profiles for bluetooth

Ah nice! How did you achieve this? Might help others with similar issues.

**Comment by lucaszanella** [comment]
I'm very confused, mine came with only 1 usb dongle, and I simply connected to 3 bluetooth devices, I didn't even use the dongle and I don't know how to use it because the 3 profiles are being used for bluetooth

**Comment by fin-ger** [comment]
Oh wow, you're right. I always thought that it is only possible to connect via bluetooth with the 3rd channel, but I just tried it and was able to connect via bluetooth with all three.

You can use the Solaar tool to pair a unifying receiver with e.g. the 2nd channel and selecting "Pair new device". Now long-press the channel-select-button on your mouse while the channel "2" is selected and you have your mouse paired with a unifying receiver.

Just to be clear: the 3 channels on the mouse are not the profiles you know from other logitech devices (at least as far as I know). Libratbag (the Piper backend) reports that the mouse does not support profiles as well as the logitech homepage is not mentioning any profile support. The 3 modes are 3 different channels to connect to for 3 different devices, e.g. your laptop, PC, and mobile phone.
